compiler/src/dmd/toobj.d

```diff
@@ -715,7 +715,8 @@ void toObjFile(Dsymbol ds, bool multiobj)
 
         override void visit(TypeInfoDeclaration tid)
         {
-            if (isSpeculativeType(tid.tinfo))
+            import dmd.globals : global;
+            if (isSpeculativeType(tid.tinfo) || global.params.betterC)
```

Comment:
I'm Not in love with this.

Should be 1 condition: does it need emission or not, rather than isSpeculative or if the compiler is in betterC.

Reply:
i.e. do the logic somewhere.else

Reply:
It could also be if `!global.params.useTypeInfo`. The problem is that the `TypeInfo` is generated by CTFE, and `DsymbolTable` gets populated with external references. 

I think a better option would be to clean it up, I tried adding a global ctfe semaphore to keep track of how deep we are in `startCTFE` (`++`) and `endCTFE`(`--`), `isInterpreting`(`++`/`--`) and actually forget about betterC by setting the `useTypeInfo` to `true` during CTFE. I was able to compile a string through `std.format` that way but the external references caused a linker problem and the application ended with the semaphore at 222.

Reply:
Actually, a betterC CTFE symbol could almost be considered speculative? Hmm